dot/rpc/websocket.go

```diff
@@ -452,15 +464,153 @@ func (l *BlockFinalizedListener) Listen() {
 		if err != nil {
 			logger.Error("failed to convert header to JSON", "error", err)
 		}
-		headM := make(map[string]interface{})
-		headM["result"] = head
-		headM["subscription"] = l.subID
 		res := newSubcriptionBaseResponseJSON()
 		res.Method = "chain_finalizedHead"
-		res.Params = headM
-		err = l.wsconn.safeSend(res)
-		if err != nil {
-			logger.Error("error sending websocket message", "error", err)
+		res.Params.Result = head
+		res.Params.SubscriptionID = l.subID
+		l.wsconn.safeSend(res)
+	}
+}
+
+// ExtrinsicSubmitListener to handle listening for extrinsic events
+type ExtrinsicSubmitListener struct {
```

Comment:
maybe this and the related logic could go into another file?

Reply:
Yes, I was thinking I should move the listener logic into another file, I'll clean-up in another PR.